src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/FailureStrategyPanel/validation.ts

```diff
@@ -124,7 +124,7 @@ export function getFailureStrategiesValidationSchema(
                 .required(getString('pipeline.failureStrategies.validation.actionRequired')),
               spec: Yup.mixed()
                 .when('type', {
-                  is: Strategy.Retry,
+                  is: val => val === Strategy.Retry || val === Strategy.RetryStepGroup,
```

Comment:
If there's a possibility for more types to get added in the future, but can be ignored for now.
```suggestion
                  is: val => [Strategy.Retry, Strategy.RetryStepGroup].includes(val)
```